lib/smb2/file.rb

```diff
@@ -58,23 +79,23 @@ def inspect
   # @param length [Fixnum,nil] number of bytes to read. If this is nil, read
   #   the whole file, i.e. {#size} bytes.
   # @return [String] full contents of the remote file
-  def read(offset: 0, length: nil)
+  def read(length = nil, offset: self.pos)
```

Comment:
I'm OK with doing `read()` and `read(length)`.  That avoids a method that does two things (seek + read).